### DISCUSSION TITLE
Is built-in tool calling a must? I'm using Llama-3.3-70b on vLLM

### DISCUSSION THREAD

**Discussion by barvhaim** [microsoft/autogen#4604] [Q&A]
Hi, I'm using vLLM without built-in tools support, is it a must have? I mean, the only implemented method is tools calling via the built-in option? 

I'm getting this error when I try to build my first agent.. how to I disable the `OpenAI style tool calls`?
```
openai.BadRequestError: Error code: 400 - {'object': 'error', 'message': '"auto" tool choice requires --enable-auto-tool-choice and --tool-call-parser to be set', 'type': 'BadRequestError', 'param': None, 'code': 400}
```

**Comment by rlam3** [comment]
@barvhaim did you get this resolved?

**Reply by barvhaim** [reply]
No, it is still an open question

**Reply by ekzhu** [reply]
For many cases built-in tool calling from model API is required for agents. However in v0.4 you can create agents with models without tool call capability, but you can't pass in tools. 

We are releasing v0.4 this week. Migration guide: https://aka.ms/autogen-migrate
